std/algorithm/mutation.d

```diff
@@ -2728,9 +2728,8 @@ Returns:
 */
 Tuple!(InputRange1, InputRange2)
 swapRanges(InputRange1, InputRange2)(InputRange1 r1, InputRange2 r2)
-    if (isInputRange!(InputRange1) && isInputRange!(InputRange2)
-            && hasSwappableElements!(InputRange1) && hasSwappableElements!(InputRange2)
-            && is(ElementType!(InputRange1) == ElementType!(InputRange2)))
+    if (hasSwappableElements!(InputRange1) && hasSwappableElements!(InputRange2)
+        && is(ElementType!(InputRange1) == ElementType!(InputRange2)))
```

Comment:
How about removing the parentheses all-together? They are probably an relict from a time where `A!B` wasn't allowed in D?